rhel/vex/parser.go

```diff
@@ -331,9 +352,59 @@ func (c *creator) lookupVulnerability(vulnKey string, protoVulnFunc func() *clai
 	return &c.fixedVulns[idx], ok
 }
 
+type ranger struct {
+	lowest map[string]*claircore.Range
+}
+
+func newRanger() *ranger {
+	return &ranger{
+		lowest: map[string]*claircore.Range{},
+	}
+}
+
+func (r *ranger) add(packageName, fixedInVersion string) (*claircore.Range, error) {
+	var rng = &claircore.Range{}
+
+	if fixedInVersion == "" {
+		zeroVer := &rhctag.Version{}
+		rng.Lower = zeroVer.Version(true)
+		highestVer := rhctag.Version{
+			Major: math.MaxInt32, // Everything is vulnerable
+		}
+		rng.Upper = highestVer.Version(true)
+	} else {
+		firstPatch, err := rhctag.Parse(fixedInVersion)
+		if err != nil {
+			return nil, err
+		}
+		rng.Upper = firstPatch.Version(false)
+		start, err := rhctag.Parse(fixedInVersion)
+		if err != nil {
+			return nil, err
```

Comment:
it'd be very weird if this managed to fail after it just succeeded a few lines above. Also a bit unfortunate you have to call the same thing twice

Reply:
Good catch, I think we can get away with just doing it once